Sources/NIO/ControlMessage.swift

```diff
@@ -172,18 +173,57 @@ struct ControlMessageParser {
     }
     
     mutating func receiveMessage(_ controlMessage: UnsafeControlMessage) {
-        if controlMessage.level == IPPROTO_IP && controlMessage.type == ControlMessageParser.ipv4TosType {
+        if controlMessage.level == IPPROTO_IP {
+            receiveIPv4Message(controlMessage)
+        } else if controlMessage.level == IPPROTO_IPV6 {
+            receiveIPv6Message(controlMessage)
```

Comment:
NIO style is to use the explicit `self` keyword when accessing member functions or properties.